File: python/src/models/donation.py
from sqlalchemy import Column, Integer, ForeignKey, BLOB, Date, Float
from sqlalchemy.orm import relationship
from database import Base


class Donation(Base):
    __tablename__ = "donation"
    id_donation = Column(Integer, primary_key=True, nullable=False, autoincrement=True)
    bloodTest = Column(BLOB)
    date = Column(Date, nullable=False)
    quantity = Column(Float)
    id_donor = Column(Integer, ForeignKey("donor.id_donor", ondelete="CASCADE"), nullable=False)
    id_request = Column(Integer, ForeignKey("request.id_request", ondelete="CASCADE"), nullable=False)
    donor = relationship("Donor", back_populates="donations")
    request = relationship("Request", back_populates="donations")

File: nodejs/database/models/donation.js
'use strict';

module.exports = (sequelize, DataTypes) => {
    const donation = sequelize.define('donation', {
        id_donation: {
            type: DataTypes.INTEGER,
            primaryKey: true,
            autoIncrement: true
        },
        bloodTest: {
            type: DataTypes.BLOB,
            allowNull: true
        },
        date: {
            type: DataTypes.DATE,
        },
        quantity: {
            type: DataTypes.FLOAT,
            allowNull: true
        },
        id_donor: {
            type: DataTypes.INTEGER
        },
        id_request: {
            type: DataTypes.INTEGER
        }
    }, {
        freezeTableName: true,
        timestamps: false,
        underscored: true,
        name: {
            singular: 'donation',
            plural: 'donation'
        }
    });

    donation.associate = models => {
        donation.belongsTo(models.donor, {
            foreignKey: 'id_donor'
        });

        donation.belongsTo(models.request, {
            foreignKey: 'id_request'
        });
    };

    return donation;
};
